org.lflang/src/org/lflang/generator/ReactionInstanceGraph.java

```diff
@@ -192,8 +199,12 @@ protected void addNodesAndEdges(ReactorInstance reactor) {
                     List<Runtime> previousRuntimes = previousReaction.getRuntimeInstances();
                     int count = 0;
                     for (Runtime runtime : runtimes) {
-                        this.addEdge(runtime, previousRuntimes.get(count));
-                        count++;
+                        // Only add the reaction order edge if previous reaction is outside of a mode or both are in the same mode
+                        // This allows modes to break cycles since modes are always mutually exclusive.
+                        if (runtime.getReaction().getMode(true) == null || runtime.getReaction().getMode(true) == reaction.getMode(true)) {
+                            this.addEdge(runtime, previousRuntimes.get(count));
```

Comment:
I'm aware that this terminology was inherited, but I find `Runtime` and "runtimes" a very strange choice... @edwardalee: how about "replicas"?

Reply:
No, "replicas" doesn't work because the vast majority of the time, there is only one of these. It's only when nested within banks that you get more than one, and even then, they aren't really replicas. They can have different parameters and serve different roles. "Replicas" makes me think of triple modular redundancy...

Reply:
OK, a better fitting term -- one that's more specific than "runtime" -- that you can think of?